linkerd/admin/src/main/scala/io/buoyant/linkerd/admin/DtabHandler.scala

```diff
@@ -0,0 +1,47 @@
+package io.buoyant.linkerd.admin
+
+import com.twitter.finagle.http.{Request, Response, Status}
+import com.twitter.finagle.{Dtab, Service}
+import com.twitter.util.Future
+import io.buoyant.linkerd.admin.names.DelegateApiHandler
+
+private[admin] class DtabHandler(
+  dtabs: () => Future[Map[String, Dtab]]
+) extends Service[Request, Response] {
+
+  import DtabHandler._
+
+  override def apply(req: Request): Future[Response] = dtabs().flatMap { dtabMap =>
+    req.path match {
+      case rexp(routerName) => dtabMap.get(routerName) match {
```

Comment:
What do you think about calling these "namespace" instead of "routerName"?  In the context of linkerd each router has one namespace so it amounts to the same thing, but using "namespace" probably makes more sense if this is to be reused by namerd.